doc/task_breakdown.md

```diff
@@ -0,0 +1,99 @@
+# SQL Query Builder: Task Breakdown
+
+This is a breakdown of tasks for building the **SQL Query Builder** application.
+
+---
+
+## 1. Interface for Building SQL Queries
+
+Table Selection:
+- A dropdown list to display available tables from the database.
+- The ability to select multiple tables at once.
+- A "Add Table" button to add selected tables to the current query.
+- A "Clear Selection" button to reset selected tables.
+
+Column Selection:
+- Checkbox lists for selecting columns from the chosen tables.
+- Display a message "Please select tables first" if no tables are selected.
+
+Adding Filters (WHERE):
+- Fields to specify conditions:
+- A dropdown list for columns.
+- A dropdown list for operators (`=`, `>`, `<`).
+- A field to input a value.
+- An "Add Filter" button to add new conditions.
+- A "Clear Filters" button to remove all set conditions.
+
+Sorting (ORDER BY):
+- A dropdown list for selecting a column to sort by.
+- The ability to specify the sorting order (ASC or DESC).
+- An "Add Sorting" button to add sorting parameters.
+- A "Clear Sorting" button to reset sorting parameters.
+
+Grouping (GROUP BY):
+- A dropdown list to select columns for grouping results.
+- Display a message "Please select columns for grouping."
+
+Having Clause:
+- Fields to specify conditions on grouped results.
+- A dropdown for columns and operators (`=`, `>`, `<`).
+- A text field to input values for conditions.
+
+---
+
+## 2. Generating and Displaying SQL Queries
+
+Generation Logic:
+- Create a function for automatically constructing the SQL query:
+- Generate `SELECT` with selected columns.
+- Generate `FROM` with selected tables.
+- Generate `WHERE` with provided filters.
+- Generate `GROUP BY` and `HAVING` if specified.
+- Generate `ORDER BY` with specified sorting parameters.
+- Handle table joins (JOIN) based on foreign keys.
+
+Viewing the Query:
+- Display the generated SQL query in a text area for review.
+- Dynamically update the query text when parameters change. 
+
+---
+
+## 3. Executing the Query and Displaying Results
+
+"Execute Query" Button:
+- Implement a button to send the generated SQL query to the server.
+
+Data Retrieval:
+- Write a method in the service to execute queries.
+- Return data from the database in JSON format.
+
+Displaying Results:
+- Display the query results in an HTML table using `<table>`, `<th>`, `<tr>`, `<td>` tags.
+- Add a "No data" message if the query result is empty.
+
+Summary for Data Structures:
+- Provide a summary view of the selected tables, columns, and conditions.
+- Display this summary dynamically based on the current query parameters.
+
+---
+
+## 4. Working with Table Joins (JOIN)
+
+Automatic Relationship Detection:
```

Comment:
How are you going to implement this task? How you can check that this is finished?

Where it should be implemented? on UI or on backend?